File: Engine/src/main/java/org/etieskrill/engine/graphics/gl/shader/impl/GridShader.java
package org.etieskrill.engine.graphics.gl.shader.impl;

import io.github.etieskrill.injection.extension.shader.reflection.ReflectShader;
import org.etieskrill.engine.graphics.gl.shader.ShaderProgram;

import java.util.List;

@ReflectShader
public class GridShader extends ShaderProgram {
    public GridShader() {
        super(List.of("Grid.glsl"));
    }

    //TODO autodetect
//    @Override
//    protected void init() {
//        hasGeometryShader();
//    }

}

File: Engine/src/commonMain/kotlin/org/etieskrill/engine/graphics/shader/impl/GridShader.kt
package org.etieskrill.engine.graphics.shader.impl

import io.github.etieskrill.injection.extension.shader.reflection.ReflectShader
import org.etieskrill.engine.graphics.shader.Shader

@ReflectShader
class GridShader : Shader(listOf("Grid.glsl")) {
    //TODO autodetect
//    @Override
//    protected void init() {
//        hasGeometryShader();
//    }
}
